mnefun/_mnefun.py

```diff
@@ -1772,3 +1777,37 @@ def gen_html_report(p, subjects, structurals, raw=True, evoked=True,
                             pattern=patterns)
         report_fname = get_report_fnames(p, subj)[0]
         report.save(report_fname, open_browser=False, overwrite=True)
+
+
+def plot_raw_psd(p, subjects, tmin=0., fmin=2, n_fft=2048):
+    """Plot data power for all available raw data files for a subject
+
+    Parameters
+    ----------
+    p : instance of Parameters
+        Analysis parameters.
+    subjects : list of str
+        Subject names to analyze (e.g., ['Eric_SoP_001', ...]).
+    tmin : float
+        Time in sec for beginning fft (defaults to 0)
+    fmin : float
+        Lower frequency edge for PSD (defaults to 2Hz)
+    Notes
+    -----
+    tmax for psd set to last time point in raw data. fmax set
+    to acquisition low pass cut off for raw and sss files, and
+    low pass cut off in analysis parameters for pca file.
+    """
+    for subj in subjects:
+        for file_type in ['raw', 'sss', 'pca']:
+            fname = get_raw_fnames(p, subj, which=file_type, erm=False)
+            if len(fname) < 1:
+                raise Warning('Unable to find %s data file.' % file_type)
```

Comment:
you don't want to `raise` because this will halt execution, you need to do `warnings.warn(...)`